### PR TITLE
Add links for ingresses

### LINKED ISSUE #4626: Allow opening or copying ingress URL
**What would you like to be added**:
Please add clickable link for each ingress final URL on both list and details :)

On ingress list page there is Name, Namespace, LoadBalancers, Rules and Age columns
Rules usually contains Ingress final URL with arrow like http://foobar.com/ -> service:80

Right now there is no easy way to open that in browser, also it's hard to copy that 
For multiple rules more than one address is visible in Rules columns, still cannot open any of them. 
On click right panel is opened, and instead of final url there is table with rules divided to host, path and backend. 

Right now for links I need to open dev tools and copy link from source (list). 

Expected changes: 
1. add link to list rules urls (that won't open details page) 
2. add same url to details somewhere near rules 
3. link backend services and tls 

**Why is this needed**:

Clickable link would speed up work a lot! 

**Environment you are Lens application on:**
win10 + k3s, but it's true for all clusters and systems.

### PR DESCRIPTION
**Pull request by Nokel81**
resolves: #4626

Screen Shot 2022-01-05 at 11 34 01 AM
Screen Shot 2022-01-05 at 12 16 36 PM

### CODE REVIEW

File: src/renderer/components/+network-ingresses/ingress-details.tsx [hunk 1]
```diff
@@ -65,44 +65,51 @@ export class IngressDetails extends React.Component<Props> {
...
+          <Table className="paths">
+            <TableHead>
+              <TableCell className="path">Path</TableCell>
+              <TableCell className="link">Link</TableCell>
+              <TableCell className="backends">Backends</TableCell>
+            </TableHead>
+            {
+              rule.http.paths
+                .map(({ path = "/", backend }, index) => {
+                  const link = `${protocol}://${rule.host || "*"}${path}`;
```
Comment (nevalla) [inline src/renderer/components/+network-ingresses/ingress-details.tsx]: if host is `*`, isn't hyperlink broken then?

Comment (Nokel81) [reply src/renderer/components/+network-ingresses/ingress-details.tsx]: Yes it is, I guess I should then not display them as links then.

**Review by aleksfront — DISMISSED** [review]

File: src/renderer/components/+network-ingresses/ingresses.tsx [hunk 2]
```diff
@@ -55,7 +55,19 @@ export class Ingresses extends React.Component<IngressesProps> {
...
+            <>
+              <a
+                key={route.url}
+                href={route.url}
+                rel="noreferrer"
+                target="_blank"
+                onClick={e => e.stopPropagation()}
+              >
+                {route.url}
+              </a> ⇢ {route.service}
```
Comment (jim-docker) [inline src/renderer/components/+network-ingresses/ingresses.tsx]: This should also not be a link if `route.url` contains `*`?

**Review by jim-docker — COMMENTED** [review]
doesn't work with resource backends? Just trying the example https://raw.githubusercontent.com/kubernetes/website/main/content/en/examples/service/networking/ingress-resource-backend.yaml from https://kubernetes.io/docs/concepts/services-networking/ingress/:

https://user-images.githubusercontent.com/40840436/158704799-a5083797-cf2c-45a5-9389-d02f9d31ad7a.mov

**Comment by Nokel81** [comment]
@jim-docker I have fixed the crash you have observed as well (even though I don't think it was caused by these changes). And here are some screenshots of how it looks now.

Screen Shot 2022-03-21 at 9 19 27 AM
Screen Shot 2022-03-21 at 9 19 17 AM

File: src/common/k8s-api/endpoints/ingress.api.ts [hunk 3]
```diff
@@ -196,6 +198,24 @@ export class Ingress extends KubeObject {
   }
 }
 
+export function computeRuleDeclarations(ingress: Ingress, rule: IngressRule): ComputedIngressRoute[] {
+  const { host = "*", http: { paths } = { paths: [] }} = rule;
+  const protocol = (ingress.spec.tls?.length ?? 0) > 0
```
Comment (jim-docker) [inline src/common/k8s-api/endpoints/ingress.api.ts]: I'm seeing every link getting https, something fishy here?

Comment (jim-docker) [reply src/common/k8s-api/endpoints/ingress.api.ts]: If I set `tls` it gets http so this is wrong, not exactly reversed with that `?? 0` I think

Comment (Nokel81) [reply src/common/k8s-api/endpoints/ingress.api.ts]: Thanks for catching this.

**Review by jim-docker — APPROVED** [review]

**Review by nevalla — APPROVED** [review]

**Pull request merged**